lib/spack/spack/installer.py

```diff
@@ -2296,7 +2301,13 @@ def next_attempt(self, installed):
     @property
     def priority(self):
         """The priority is based on the remaining uninstalled dependencies."""
-        return len(self.uninstalled_deps)
+        priority = len(self.uninstalled_deps)
+
+        # If there are no uninstalled dependencies then the priority should
+        # be a random number in [0, 1) to allow parallel installs a chance
+        # of more quickly finding the next package to install; otherwise,
+        # the priority will remain the number of uninstalled dependencies.
+        return random.random() if priority == 0 else priority
```

Comment:
Wouldn't it be better to generate the random number once ahead of time in the constructor or so, and order by `(#retries, self.random_number)`? Suppose you have 2 tasks and one is busy, then if you randomly select the busy task, it would make sense to always try the other task instead of maybe randomly selecting the busy one another time.

Reply:
Well, something like that.  The key throws a monkey wrench into the works.  So I tried a variant of this approach that I currently think will have the same affect.